app/src/main/java/org/eyeseetea/malariacare/data/di/Injector.java

```diff
@@ -0,0 +1,100 @@
+package org.eyeseetea.malariacare.data.di;
+
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.joda.JodaModule;
+
+import org.eyeseetea.malariacare.BuildConfig;
+import org.eyeseetea.malariacare.data.database.model.OptionDB;
+import org.eyeseetea.malariacare.data.database.model.QuestionDB;
+import org.eyeseetea.malariacare.data.sync.importer.poeditor.POEditorApiClient;
+import org.eyeseetea.malariacare.data.sync.importer.strategies.ILanguagesClient;
+import org.eyeseetea.malariacare.data.sync.importer.strategies.LanguageDownloader;
+import org.eyeseetea.malariacare.domain.boundary.IConnectivityManager;
+import org.eyeseetea.malariacare.domain.boundary.converters.IConverter;
+import org.eyeseetea.malariacare.data.database.converts.OptionConverterFromDomainModelToDB;
+import org.eyeseetea.malariacare.data.database.converts.QuestionConverterFromDomainModelToDB;
+import org.eyeseetea.malariacare.domain.entity.Option;
+import org.eyeseetea.malariacare.domain.entity.Question;
+import org.eyeseetea.malariacare.utils.ConnectivityStatus;
+
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+
+public class Injector {
```

Comment:
I'm a bit concerned about introducing this "Injector", that's even in a "di" package as if we were trying to use it as a dependency injector. I'm not sure here and I would like to know @xurxodev's opinion. Could we just create factories (that's something we've already used in the app before)? At least, this element is something that we should discuss with @xurxodev, that for sure.

Reply:
@ifoche I have looked this class a while, really this class is not a injector is a factory. I think we should rename this class and its package